internal/remediation/suggester/maven.go

```diff
@@ -2,23 +2,124 @@ package suggester
+			Type:       req.Type,
+			NewRequire: latest.Version,
+		}
+
+		origReq := originalRequirement(req, specific.OriginalImports)
+		if !requirementHasProperty(origReq) {
+			// The original requirement does not contain a property placeholder.
+			changedDeps = append(changedDeps, patch)
+			continue
+		}
+
+		patches, ok := generatePropertyPatches(origReq, latest.Version)
+		if !ok {
+			// Not able to update properties to update the requirement.
+			// Update the dependency directly instead.
+			changedDeps = append(changedDeps, patch)
+			continue
+		}
+
+		depOrigin, _ := req.Type.GetAttr(dep.MavenDependencyOrigin)
+		if strings.HasPrefix(depOrigin, manifest.OriginProfile) {
+			// Dependency management is not indicated in property origin.
+			depOrigin, _ = strings.CutSuffix(depOrigin, "@"+manifest.OriginManagement)
+		} else {
+			// Properties are defined either universally or in a profile. For property
+			// origin not starting with 'profile', this is an universal property.
+			depOrigin = ""
+		}
+
+		for name, value := range patches {
+			// A dependency in a profile may contain properties from this profile or
+			// properties universally defined. We need to figure out the origin of these
+			// properties. If a property is defined both universally and in the profile,
+			// we use the profile's origin.
+			propertyOrigin := ""
+			for _, p := range specific.Properties {
+				if p.Name == name && p.Origin != "" && p.Origin == depOrigin {
+					propertyOrigin = depOrigin
+				}
+			}
+			if _, ok := propertyPatches[propertyOrigin]; !ok {
+				propertyPatches[propertyOrigin] = make(map[string]string)
+			}
+			// This property has been set to update to a value. If both values are the
+			// same, we do nothing; otherwise, instead of updating the property, we
+			// should update the dependency directly.
+			if preset, ok := propertyPatches[propertyOrigin][name]; !ok {
+				propertyPatches[propertyOrigin][name] = value
+			} else if preset != value {
+				changedDeps = append(changedDeps, patch)
+			}
+		}
+	}
+
+	return manifest.ManifestPatch{
+		Deps:              changedDeps,
+		EcosystemSpecific: propertyPatches,
+	}, nil
+}
+
+// suggestVersion returns the latest version based on the given Maven requirement version.
 // If there is no newer version available, req will be returned.
 // For a version range requirement,
 //   - the greatest version matching the constraint is assumed when deciding whether the
 //     update is a major update or not.
 //   - if the latest version does not satisfy the constraint, this version is returned;
 //     otherwise, the original version range requirement is returned.
-func (ms *MavenSuggester) Suggest(ctx context.Context, client resolve.Client, req resolve.RequirementVersion, opts SuggestOptions) (resolve.RequirementVersion, error) {
+func suggestVersion(ctx context.Context, client resolve.Client, req resolve.RequirementVersion, noMajorUpdates bool) (resolve.RequirementVersion, error) {
```

Comment:
nit: `suggestVersion` might get confusing when we add other ecosystems to this suggester submodule.

Maybe make this something like `func suggestMavenVersion()` or `func (ms *MavenSuggester) suggestVersion()`.

Reply:
done!